### DISCUSSION TITLE
issue with connection pool ?

### DISCUSSION THREAD

**Discussion by mdespriee** [prisma/prisma#7642] [Q&A] [answered]
Hello,

I'm trying to diagnose a connection pool problem we have in production.
The DB is a timescaleDB, but viewed from Prisma, it's simply a postgresql. 
(The DB is hosted at a DBaaS provider, and we don't have as much logs and diag as we would like unfortunately).

The symptom is the following: after a few hours of uptime, we start to see these errors growing:
```
Timed out fetching a new connection from the pool. Please consider reducing the number of requests or increasing the `connection_limit` parameter (...). Current limit: 25, amount of connections currently in use: 16.
```
First, I don't see why there's a timeout fetching a connection, since the max is not reached.
And then the connection pool seems already pretty oversized to me: 600 query/mn, ~500ms per query in avg, and we have 2 parallel instances. So 25 should be plenty.

The problem just worsen and worsen until we reboot the service, and everything goes back to normal.

It looks like some connections are not re-used, as if they were waiting for the server or something. What can I do, application-side, to better diagnose the problem ?

**Comment by ryands17** [comment, marked as answer]
@mdespriee 👋 

It would be great if you could create an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with all the required info so that we can look into this.

I assume you're using just a single instance of `PrismaClient` across the entire project. Also Serverless environments sometimes show this behaviour so we would like to know more about the environment you're running as well :)

**Reply by mdespriee** [reply]
done https://github.com/prisma/prisma/issues/7644
it's a docker-based deployment, and 1 prisma client instance, yes.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
